packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts

```diff
@@ -1031,6 +1031,35 @@ integTest(
   }),
 );
 
+integTest(
+  'cdk diff with large changeset does not fail',
+  withDefaultFixture(async (fixture) => {
+    // GIVEN - small initial stack with only ane IAM role
+    await fixture.cdkDeploy('iam-roles', {
+      modEnv: {
+        NUMBER_OF_ROLES: '1',
+      },
+    });
+
+    // WHEN - adding 200 roles to the same stack to create a large diff
+    const diff = await fixture.cdk(['diff', fixture.fullStackName('iam-roles')], {
+      verbose: true,
+      modEnv: {
+        NUMBER_OF_ROLES: '200',
+      },
+    });
+
+    // Assert that S3 upload and changeset creation error messages were not thrown:
+    expect(diff).not.toContain('Call failed: getBucketLocation');
+    expect(diff).not.toContain('not authorized to perform: s3:GetBucketLocation on resource');
+    expect(diff).not.toContain('Could not create a change set, will base the diff on template differences');
```

Comment:
This is not very robust testing. You're better off testing against something that you expect to see, vs something that you don't expect to see. There are a million things that could go wrong to make not exactly this error message show up.